subsys/bluetooth/audio/cap_commander.c

```diff
@@ -77,17 +81,240 @@ int bt_cap_commander_discover(struct bt_conn *conn)
+			return false;
+		}
+	}
+
+	return true;
+}
+
 int bt_cap_commander_broadcast_reception_start(
 	const struct bt_cap_commander_broadcast_reception_start_param *param)
 {
-	return -ENOSYS;
+	const struct bt_cap_commander_proc_param *proc_param;
+	struct bt_cap_common_proc *active_proc;
+	struct bt_conn *conn;
+	struct bt_bap_broadcast_assistant_add_src_param add_src_param = {0};
+	int err;
+
+	if (bt_cap_common_proc_is_active()) {
+		LOG_DBG("A CAP procedure is already in progress");
+
+		return -EBUSY;
+	}
+
+	if (!valid_broadcast_reception_start_param(param)) {
+		return -EINVAL;
+	}
+
+	bt_cap_common_start_proc(BT_CAP_COMMON_PROC_TYPE_BROADCAST_RECEPTION_START, param->count);
+
+	broadcast_assistant_cb.add_src = cap_commander_ba_add_src_cb;
+	if (!ba_cb_registered && cap_commander_register_ba_cb() != 0) {
+		LOG_DBG("Failed to register VCP callbacks");
+
+		return -ENOEXEC;
+	}
+
+	active_proc = bt_cap_common_get_active_proc();
+
+	for (size_t i = 0U; i < param->count; i++) {
+		const struct bt_cap_commander_broadcast_reception_start_member_param
+			*member_param = &param->param[i];
+		struct bt_conn *member_conn =
+			bt_cap_common_get_member_conn(param->type, &member_param->member);
+
+		if (member_conn == NULL) {
+			LOG_DBG("Invalid param->members[%zu]", i);
+
+			return -EINVAL;
+		}
+
+		/* Store the necessary parameters as we cannot assume that the supplied parameters
+		 * are kept valid
+		 * TODO: consider putting this into a function
+		 */
+		active_proc->proc_param.commander[i].conn = member_conn;
+		bt_addr_le_copy(
+			&active_proc->proc_param.commander[i].broadcast_reception_start.addr,
+			&member_param->addr);
+		active_proc->proc_param.commander[i].broadcast_reception_start.adv_sid =
+			member_param->adv_sid;
+		active_proc->proc_param.commander[i].broadcast_reception_start.broadcast_id =
+			member_param->broadcast_id;
+		active_proc->proc_param.commander[i].broadcast_reception_start.pa_interval =
+			member_param->pa_interval;
+		active_proc->proc_param.commander[i].broadcast_reception_start.num_subgroups =
+			member_param->num_subgroups;
+		active_proc->proc_param.commander[i].broadcast_reception_start.subgroups =
+			member_param->subgroups;
+	}
+
+	active_proc->proc_initiated_cnt++;
+
+	proc_param = &active_proc->proc_param.commander[0];
+
+	conn = proc_param->conn;
+	bt_addr_le_copy(&add_src_param.addr, &proc_param->broadcast_reception_start.addr);
+	add_src_param.adv_sid = proc_param->broadcast_reception_start.adv_sid;
+	add_src_param.broadcast_id = proc_param->broadcast_reception_start.broadcast_id;
+	add_src_param.pa_interval = proc_param->broadcast_reception_start.pa_interval;
+	add_src_param.num_subgroups = proc_param->broadcast_reception_start.num_subgroups;
+	add_src_param.subgroups = proc_param->broadcast_reception_start.subgroups;
+
+	/* TODO: what to do if we are modifying a source that has already been added? */
+	err = bt_bap_broadcast_assistant_add_src(conn, &add_src_param);
```

Comment:
Comment doesn't really match the code, as we are doing `add_src` here and not `mod_src`.